src/main/java/picard/sam/ValidateSamFile.java

```
@@ -133,75 +133,122 @@ public static void main(final String[] args) {
+            }
+            if (VALIDATE_INDEX) {
+                validator.setIndexValidationStringency(VALIDATE_INDEX ? IndexValidationStringency.EXHAUSTIVE : IndexValidationStringency.NONE);
+            }
+            if (IOUtil.isRegularPath(INPUT)) {
+                // Do not check termination if reading from a stream
+                validator.validateBamFileTermination(INPUT);
+            }
 
-        result = false;
+            result = false;
 
-        switch (MODE) {
-            case SUMMARY:
-                result = validator.validateSamFileSummary(samReader, reference);
-                break;
-            case VERBOSE:
-                result = validator.validateSamFileVerbose(samReader, reference);
-                break;
+            switch (MODE) {
+                case SUMMARY:
+                    result = validator.validateSamFileSummary(samReader, reference);
+                    break;
+                case VERBOSE:
+                    result = validator.validateSamFileVerbose(samReader, reference);
+                    break;
+            }
+            out.flush();
+
+            if (result) {
+                return ReturnTypes.SUCCESSFUL.value();  // ran successfully with no warnings or errors
+            } else {
+                if (validator.getNumErrors() == 0) {
+                    if (validator.getNumWarnings() > 0) {
+                        return ReturnTypes.WARNINGS.value();   // warnings but no errors
+                    }
+                } else {
+                    if (validator.getNumWarnings() > 0) {
+                        return ReturnTypes.ERRORS_WARNINGS.value();   // errors and warnings
+                    } else {
+                        return ReturnTypes.ERRORS.value();  // errors but no warnings
+                    }
+                }
+            }
+        } catch (Exception e) {
+            return ReturnTypes.FAILED.value();  // failed to complete execution
```

you eat the exception here which will make the output very annoying. 

Can you log the exception message via the logger?